server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1088,8 +1188,75 @@ protected void parseCreateField(ParseContext context) throws IOException {
         }
     }
 
+    @Override
+    public CheckedConsumer<IndexTimeScriptContext, IOException> getPostParsePhase() {
+        if (this.script == null) {
+            return null;
+        }
+        return c -> this.script.compiledScript.executeAndIndex(
+            c.searchLookup,
+            c.leafReaderContext,
+            0,
+            v -> this.indexValue(c.pc, v)
+        );
+    }
+
     @Override
     public FieldMapper.Builder getMergeBuilder() {
         return new Builder(simpleName(), type, ignoreMalformedByDefault, coerceByDefault).init(this);
     }
+
+    private interface NumberMapperScript {
+        void executeAndIndex(SearchLookup lookup, LeafReaderContext ctx, int doc, Consumer<Number> emitter);
+    }
+
+    private static class LongMapperScript implements NumberMapperScript {
+
+        final LongFieldScript.Factory scriptFactory;
+        final Map<String, Object> scriptParams;
+        final String fieldName;
+
+        private LongMapperScript(String fieldName, LongFieldScript.Factory scriptFactory, Map<String, Object> scriptParams) {
+            this.scriptFactory = scriptFactory;
+            this.scriptParams = scriptParams;
+            this.fieldName = fieldName;
+        }
+
+        @Override
+        public void executeAndIndex(SearchLookup lookup, LeafReaderContext ctx, int doc, Consumer<Number> emitter) {
+            LongFieldScript s = scriptFactory
+                .newFactory(fieldName, scriptParams, lookup)
+                .newInstance(ctx);
+            s.runForDoc(doc);
+            long[] vs = s.values();
+            for (int i = 0; i < s.count(); i++) {
+                emitter.accept(vs[i]);
+            }
+        }
+    }
+
+    private static class DoubleMapperScript implements NumberMapperScript {
+
+        final DoubleFieldScript.Factory scriptFactory;
+        final Map<String, Object> scriptParams;
+        final String fieldName;
+
+        private DoubleMapperScript(String fieldName, DoubleFieldScript.Factory scriptFactory, Map<String, Object> scriptParams) {
+            this.scriptFactory = scriptFactory;
+            this.scriptParams = scriptParams;
+            this.fieldName = fieldName;
+        }
+
+        @Override
+        public void executeAndIndex(SearchLookup lookup, LeafReaderContext ctx, int doc, Consumer<Number> emitter) {
+            DoubleFieldScript s = scriptFactory
+                .newFactory(fieldName, scriptParams, lookup)
+                .newInstance(ctx);
+            s.runForDoc(doc);
+            double[] vs = s.values();
+            for (int i = 0; i < s.count(); i++) {
+                emitter.accept(vs[i]);
+            }
```

Comment:
can't help but wonder if these two impls could share more of their code. I guess that it all starts with the factories not sharing a base class, and I vaguely remember talking with @nik9000 about this not being straight-forward...

Reply:
Yes, I tried this, but it's the same problem as with the various runtime field types - if we want to avoid boxing then everything needs to have a separate implementation to handle primitive types.

Reply:
I think if we shared the `scriptParams` and `fieldName` with a superclass and did the thing I was talking about in the comment above these classes would be very very close to one liners. As close as you can be in java, at any rate.

Reply:
I've collapsed them both down into anonymous classes returned by their relative numeric types.  Nik's idea will make them even smaller but let's maybe do that later?